### PR TITLE
Variables defined in JSON

### PR DESCRIPTION
**Pull request by bear454**

### CODE REVIEW

File: aks/README.md [hunk 1]
```diff
@@ -44,7 +45,7 @@
       nginx.ingress.kubernetes.io/proxy-body-size: 1024m
       certmanager.k8s.io/cluster-issuer: letsencrypt-prod
       certmanager.k8s.io/acme-challenge-type: dns01
-      certmanager.k8s.io/acme-dns01-provider: azuredns
+      certmanager.k8s.io/ace-dns01-provider: azuredns
```
Comment (satadruroy) [inline aks/README.md]: Typo perhaps? This has to be `acme` as in the https://en.wikipedia.org/wiki/Automated_Certificate_Management_Environment protocol.

File: aks/terraform.tfvars.json [hunk 2]
```diff
@@ -1,15 +1,15 @@
 {
   "location": "xxxxxxxxxxxxxxxxxxxxxxxxx",
-  "az_resource_group": "xxxxxxxxxxxxxxxxxxxxxxxxx",
+  "resource_group": "xxxxxxxxxxxxxxxxxxxxxxxxx",
   "ssh_public_key": "/xxxxxxxxxxxxxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxx.pub",
-  "agent_admin": "xxxxxxxxxxxxxxxxxxxxxxxxx",
+  "ssh_username": "xxxxxxxxxxxxxxxxxxxxxxxxx",
```
Comment (satadruroy) [inline aks/terraform.tfvars.json]: I assume the ssh public key is not being sourced from the file-system and the actual contents of the public key are expected here?

File: aks/terraform.tfvars.json [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
   "client_id": "xxxxxxxxxxxxxxxxxxxxxxxxx",
   "client_secret": "xxxxxxxxxxxxxxxxxxxxxxxxx",
   "cluster_labels": {},
   "azure_dns_json": "/xxxxxxxxxxxxxxxxxxxxxxxxx/xxxxxxxxxxxxxxxxxxxxxxxxx/azure_dns_json.json",
```
Comment (satadruroy) [inline aks/terraform.tfvars.json]: You probably don't want to source the `external-dns` azure config file from the filesystem. Instead, you can provide them to the helm deployment for external-dns as documented here:

https://github.com/helm/charts/tree/master/stable/external-dns

(check the azure parameters for resource group, tenant id, subscription id etc.)

File: aks/terraform.tfvars.json (hunk 2 above)
Comment (bear454) [reply aks/terraform.tfvars.json]: This file is cruft; removed in 1e04545 .

File: aks/terraform.tfvars.json (hunk 3 above)
Comment (bear454) [reply aks/terraform.tfvars.json]: This file is cruft; removed in 1e04545 .

File: aks/README.md (hunk 1 above)
Comment (bear454) [reply aks/README.md]: Fixed in 46fb21b .

**Pull request merged**